File: Php/tests/CoreTests/GravatarTests.php
<?php
namespace tests\CoreTests{

	use core\ClassNames;
	use integration\Avatar\GravatarService;
	use tests\TestHelper;
	use tests\UnitTestCase;

	class GravatarTests extends UnitTestCase {

		private $settings;

		function setUp(){
			$this->settings = TestHelper::getFake(ClassNames::$Settings);
		}

		function testSmallGravatarUrl(){
			$gravatarEmail = "henriks@gmail.com";
			$this->settings->returns('getSiteUrl', 'site-url');
			$expectedUrl = "http://www.gravatar.com/avatar/24a827c683a7646cde86696b418b20b4?s=40&d=site-url/core/ui/img/pix.gif";
			$gravatarService = $this->getGravatarService();

			$gravatarUrl = $gravatarService->getSmallAvatarUrl($gravatarEmail);
			$this->assertIdentical($expectedUrl, $gravatarUrl);
		}

		function testLargeGravatarUrl(){
			$gravatarEmail = "henriks@gmail.com";
			$this->settings->returns('getSiteUrl', 'site-url');
			$expectedUrl = "http://www.gravatar.com/avatar/24a827c683a7646cde86696b418b20b4?s=100&d=site-url/core/ui/img/pix.gif";
			$gravatarService = $this->getGravatarService();

			$gravatarUrl = $gravatarService->getLargeAvatarUrl($gravatarEmail);
			$this->assertIdentical($expectedUrl, $gravatarUrl);
		}

		private function getGravatarService(){
			return new GravatarService($this->settings);
		}

	}

}

File: Infrastructure.Tests/Gravatar/GravatarTests.cs
using Infrastructure.Integration.Gravatar;
using NUnit.Framework;
using Tests.Common;

namespace Infrastructure.Tests.Gravatar{

	public class GravatarTests : MockContainer {

	    private const string TestEmail = "henriks@gmail.com";
	    private const string TestHash = "abcdef";

	    [Test]
		public void SmallGravatarUrl(){
	        const string expectedUrlFormat = "http://www.gravatar.com/avatar/{0}?s=40&d=site-url/FrontEnd/Images/pix.gif";
	        var expected = string.Format(expectedUrlFormat, TestHash);
	        EncryptionServiceMock.Setup(o => o.GetMd5Hash(TestEmail)).Returns(TestHash);
            SettingsMock.Setup(o => o.GetSiteUrl()).Returns("site-url");
			
            var sut = GetSut();

			var result = sut.GetSmallAvatarUrl(TestEmail);
			Assert.AreEqual(expected, result);
		}

        [Test]
		public void LargeGravatarUrl(){
			const string expectedUrlFormat = "http://www.gravatar.com/avatar/{0}?s=100&d=site-url/FrontEnd/Images/pix.gif";
            var expected = string.Format(expectedUrlFormat, TestHash);
            EncryptionServiceMock.Setup(o => o.GetMd5Hash(TestEmail)).Returns(TestHash);
			SettingsMock.Setup(o => o.GetSiteUrl()).Returns("site-url");
            
            var sut = GetSut();

            var result = sut.GetLargeAvatarUrl(TestEmail);
			Assert.AreEqual(expected, result);
		}

		private GravatarService GetSut(){
			return new GravatarService(SettingsMock.Object, EncryptionServiceMock.Object);
		}

	}

}